### PR TITLE
Capella block support

### PR DESCRIPTION
**Pull request by olegshmuelov**
- add Capella block/blinded block tests
- switch to use VersionedValidatorRegistration
- adjust retrieving slashing protection data from DB

### CODE REVIEW

File: slashing_protection/normal_protection.go [hunk 1]
```diff
@@ -106,15 +106,14 @@ func (protector *NormalProtection) UpdateHighestProposal(key []byte, slot phase0
...
-		if err != nil {
+	if highest == nil {
+		if err := protector.store.SaveHighestProposal(key, slot); err != nil {
 			return errors.New("could not save highest proposal")
 		}
 		return nil
 	}
 
-	if highest < slot {
+	if *highest < slot {
```
Comment (moshe-blox) [inline slashing_protection/normal_protection.go:116]: Could be simplified to?
```go
if highest == nil || *highest < slot {
		return protector.store.SaveHighestProposal(key, slot)
}
```

**Review by moshe-blox — APPROVED** [review]

**Comment by moshe-blox** [comment]
Remove ssv-spec dependency (and maybe copy the func we used into ekm [instead of importing])

**Review by moshe-blox — APPROVED** [review]

**Review by lior-blox — APPROVED** [review]

File: signer/sign_epoch.go [hunk 2]
```diff
@@ -23,7 +22,7 @@ func (signer *SimpleSigner) SignEpoch(epoch phase0.Epoch, domain phase0.Domain,
 		return nil, nil, err
 	}
 
-	root, err := types.ComputeETHSigningRoot(types.SSZUint64(epoch), domain)
+	root, err := ComputeETHSigningRoot(SSZUint64(epoch), domain)
```
Comment (olegshmuelov) [inline signer/sign_epoch.go:25]: add test for ComputeETHSigningRoot

File: slashing_protection/normal_protection.go [hunk 3]
```diff
@@ -47,11 +47,11 @@ func (protector *NormalProtection) IsSlashableProposal(pubKey []byte, slot phase
 	if err != nil {
 		return nil, errors.New("could not retrieve highest proposal")
 	}
-	if highest == 0 {
+	if highest == nil {
 		return nil, errors.New("highest proposal data is nil, can't determine if proposal is slashable")
 	}
 
-	if slot > highest {
+	if slot > *highest {
```
Comment (olegshmuelov) [inline slashing_protection/normal_protection.go:54]: revert to use dereference

**Review by moshe-blox — APPROVED** [review]

**Review by lior-blox — APPROVED** [review]

**Pull request merged**